source/server/config/network/ext_authz.cc

```diff
@@ -0,0 +1,69 @@
+#include "server/config/network/ext_authz.h"
+
+#include <chrono>
+#include <string>
+
+#include "envoy/config/filter/network/ext_authz/v2/ext_authz.pb.validate.h"
+#include "envoy/ext_authz/ext_authz.h"
+#include "envoy/network/connection.h"
+#include "envoy/registry/registry.h"
+
+#include "common/ext_authz/ext_authz_impl.h"
+#include "common/filter/ext_authz.h"
+#include "common/protobuf/utility.h"
+
+namespace Envoy {
+namespace Server {
+namespace Configuration {
+
+NetworkFilterFactoryCb ExtAuthzConfigFactory::createFilter(
+    const envoy::config::filter::network::ext_authz::v2::ExtAuthz& proto_config,
+    FactoryContext& context) {
+
+  ASSERT(!proto_config.stat_prefix().empty());
+  ASSERT(!proto_config.grpc_service().envoy_grpc().cluster_name().empty());
+
+  ExtAuthz::TcpFilter::ConfigSharedPtr ext_authz_config(
+      new ExtAuthz::TcpFilter::Config(proto_config, context.scope()));
+  const uint32_t timeout_ms = PROTOBUF_GET_MS_OR_DEFAULT(proto_config.grpc_service(), timeout, 200);
+
+  return [ grpc_service = proto_config.grpc_service(), &context, ext_authz_config,
+           timeout_ms ](Network::FilterManager & filter_manager)
+      ->void {
+
+    auto async_client_factory =
+        context.clusterManager().grpcAsyncClientManager().factoryForGrpcService(grpc_service,
+                                                                                context.scope());
+
+    auto client = std::make_unique<Envoy::ExtAuthz::GrpcClientImpl>(
+        async_client_factory->create(), std::chrono::milliseconds(timeout_ms));
+    filter_manager.addReadFilter(Network::ReadFilterSharedPtr{
+        new ExtAuthz::TcpFilter::Instance(ext_authz_config, std::move(client))});
+  };
+}
+
+NetworkFilterFactoryCb ExtAuthzConfigFactory::createFilterFactory(const Json::Object& json_config,
```

Comment:
We only need this for v1 config; are we supporting v1 ext_authz?

Reply:
we are doing only v2. This helps in specifying the configuration as v2 json and still use it without rigging up a xDS server.

Reply:
Not sure this is strictly needed for v2 JSON, this should only apply to v1 JSON placed in `deprecated_v1`. Does the test fail if you don't have this?

Reply:
afaik this function is the entry point when I give envoy executable a configuration file `envoy.conf` and I can then specify the filter's configuration like:
```
      "filters": [
        {
         "type": "read",
         "name": "ext_authz",
         "config": { 
                     "stat_prefix": "authz",
                     "grpc_service": {
                        "envoy_grpc": { "cluster_name": "authz_server" }
                      },
                     "failure_mode_allow": true
                   }
        },
```
And that way I can run an end-2-end system.
i'll remove it since it seems to be not what Envoy natively expects to support.